gobblin-compaction/src/main/java/gobblin/compaction/Dataset.java

```diff
@@ -17,12 +17,15 @@
 import java.util.List;
 import java.util.concurrent.atomic.AtomicReference;
 
+import org.apache.commons.lang.StringUtils;
 import org.apache.hadoop.fs.Path;
 
+import com.google.common.base.Optional;
 import com.google.common.collect.Lists;
 
 import lombok.extern.slf4j.Slf4j;
 
+import gobblin.compaction.mapreduce.MRCompactor;
```

Comment:
Gobblin compaction dataset is different from the `gobblin.data.management.dataset.Dataset`. It contains the `inputPath` and `outputPath`, which could be different and not suitable for `gobblin.data.management.dataset.Dataset#datasetRoot()`.